frb_rust/src/support.rs

```diff
@@ -50,3 +50,52 @@ pub struct WireSyncReturnStruct {
     pub len: i32,
     pub success: bool,
 }
+
+impl From<WireSyncReturnData> for WireSyncReturnStruct {
+    fn from(data: WireSyncReturnData) -> Self {
+        let (ptr, len) = into_leak_vec_ptr(data.0);
+        WireSyncReturnStruct {
+            ptr,
+            len,
+            success: true,
+        }
+    }
+}
+
+/// Wrapper struct for [`WireSyncReturnStruct`].
```

Comment:
nit: Seems the comment can be improved a bit, since "wrapper of A" looks like "`struct ThisStruct{inner: A}" to me.

What about saying "intermediate" or "safe version of WireSyncReturnStruct"